### PR TITLE
tachyon: disable threading on aarch64 to avoid sage test hangs

### PR DESCRIPTION
**Pull request by collares**
###### Motivation for this change

Something changed on master between [March 11th](https://github.com/NixOS/nixpkgs/pull/115980) and [March 20th](https://github.com/NixOS/nixpkgs/pull/117017) and Sage tests that use Tachyon for 3D graphics started hanging on aarch64 (the links are to succeeding and failing ofBorg testruns). I don't have an aarch64 machine so it's hard for me to bisect the problem, but disabling multithreading makes the bug go away.

@timokau, is this an acceptable solution?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by collares** [comment]
@ofborg build sage

**Comment by collares** [comment]
@timokau I am not in a hurry at all to get this merged as I don't even have access to an aarch64 system, but I thought I'd send a "this is ready (for feedback, at least)" ping just because you probably got a billion emails from my force pushes while this PR was still a draft. This fixes Sage aarch64 tests on ofBorg.

**Review by 7c6f434c — APPROVED** [review]

File: pkgs/development/libraries/tachyon/default.nix [hunk 1]
```diff
@@ -36,8 +36,8 @@ stdenv.mkDerivation rec {
   '';
   arch = if stdenv.hostPlatform.system == "x86_64-linux"   then "linux-64-thr"  else
          if stdenv.hostPlatform.system == "i686-linux"     then "linux-thr"     else
-         if stdenv.hostPlatform.system == "aarch64-linux"  then "linux-arm-thr" else
-         if stdenv.hostPlatform.system == "armv7l-linux"   then "linux-arm-thr" else
+         if stdenv.hostPlatform.system == "aarch64-linux"  then "linux-arm"     else
```
Comment (timokau) [inline pkgs/development/libraries/tachyon/default.nix:39]: Please add a comment that explains why threading is disabled. You could just link to this thread. That might be helpful when somebody wants to work on the derivation in the future.

**Review by timokau — CHANGES_REQUESTED** [review]
I tried to bisect the issue, but I already got test timeouts on aarch64 right after the sage 9.2 update (#105615). The aarch64 support of sage on nix was never exactly first class, so this might go back even further.

Thank you for tracking it down. I left a comment requesting some "breadcrumbs" for future adventurers. After that this PR should be good to go.

File: pkgs/development/libraries/tachyon/default.nix (hunk 1 above)
Comment (collares) [reply pkgs/development/libraries/tachyon/default.nix:42]: Good idea, I've added a comment. Thanks!

**Review by timokau — APPROVED** [review]

**Pull request merged**